app_legacy/Helpers/database/user-activity.php

```diff
@@ -500,15 +500,15 @@ function getLatestRichPresenceUpdates(): array
     $recentMinutes = 10;
     $permissionsCutoff = Permissions::Registered;
 
-    $query = "SELECT ua.User, ua.RAPoints, ua.RASoftcorePoints, ua.RichPresenceMsg,
-                     gd.ID AS GameID, gd.Title AS GameTitle, gd.ImageIcon AS GameIcon, c.Name AS ConsoleName
+    $query = "SELECT ua.User, IF(ua.Untracked, 0, ua.RAPoints) as RAPoints, IF(ua.Untracked, 0, ua.RASoftcorePoints) as RASoftcorePoints, 
+                     ua.RichPresenceMsg, gd.ID AS GameID, gd.Title AS GameTitle, gd.ImageIcon AS GameIcon, c.Name AS ConsoleName
               FROM UserAccounts AS ua
               LEFT JOIN GameData AS gd ON gd.ID = ua.LastGameID
               LEFT JOIN Console AS c ON c.ID = gd.ConsoleID
               WHERE ua.RichPresenceMsgDate > TIMESTAMPADD( MINUTE, -$recentMinutes, NOW() )
-                AND ua.LastGameID !=0
+                AND ua.LastGameID != 0
                 AND ua.Permissions >= $permissionsCutoff
-              ORDER BY ua.RAPoints DESC";
+              ORDER BY RAPoints DESC";
```

Comment:
I think we need to modify the sort here. The current sort just prioritizes hardcore users, then everyone else is just kind of randomly at the bottom.

At a minimum, there should probably be a secondary sort on ua.RASoftcorePoints so the untracked users don't appear in the middle of the softcore user list. I don't know if we need an explicit third-level sort for the untracked users.

Reply:
To be sure I am completely following, do you think a change like:

```sql
ORDER BY RAPoints DESC, RASoftcorePoints DESC
```

would satisfy this and make the sorting more meaningful? 

I am not sure if we need an explicit third-level sort for the untracked users because the `IF(ua.Untracked, 0, ua.RAPoints)` and `IF(ua.Untracked, 0, ua.RASoftcorePoints) `conditions in the query already handle untracked users by setting their RAPoints and RASoftcorePoints to 0. 

This means the ORDER BY in the snippet above will already place untracked users at the bottom of the list, causing them to naturally appear after all the tracked users regardless of whether they are hardcore or softcore players.

Reply:
Yes to `ORDER BY RAPoints DESC, RASoftcorePoints DESC`. That would guarantee a separation between softcore players and untracked users, _and_ prioritize softcore players with more points in a similar fashion to the way the main sort prioritizes hardcore players with more points.

My thought is that all untracked users would have 0/0 points, so they would just be at the end of the list with no guaranteed sort. I don't think it's a big deal, just noting the behavior.

Reply:
I understand; I've opted to do:

```sql
ORDER BY RAPoints DESC, RASoftcorePoints DESC, ua.User ASC
```

Now those users floating around the bottom should at least be alphabetically sorted A-Z by their usernames, giving us a bit more determinism in the ultimate ordering of these entities. This change is reflected in my latest commit.